From 6d67118bbf44c939f82eceb4e50e039f48faeb93 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 7 Aug 2026 03:17:53 +0000
Subject: [PATCH 33/38] swap (Apple): don't gate vm.compressor_mode check on
 macOS 26 availability

__builtin_available(macOS 26.0, *) isn't meaningful on legacy SDKs;
comment out the branch so the vm.compressor_mode check always runs.

Ported from https://github.com/macos-powerpc/powerpc-ports/blob/main/sysutils/fastfetch/files/0010-swap_fat_arch_64-does-not-exist-in-10.6.patch
---
 src/detection/swap/swap_apple.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/detection/swap/swap_apple.c b/src/detection/swap/swap_apple.c
index 3c457c0ec..b628b042e 100644
--- a/src/detection/swap/swap_apple.c
+++ b/src/detection/swap/swap_apple.c
@@ -11,13 +11,13 @@ const char* ffDetectSwap(FFlist* result) {
     }
 
     if (xsw.xsu_total == 0) {
-        if (__builtin_available(macOS 26.0, *)) {
+//        if (__builtin_available(macOS 26.0, *)) {
             // "vm.compressor_mode" no longer exists in macOS 26.0
-        } else {
+//        } else {
             if (ffSysctlGetInt("vm.compressor_mode", 4) <= 2) {
                 return nullptr; // Swap is disabled
             }
-        }
+//        }
     }
 
     FFSwapResult* swap = FF_LIST_ADD(FFSwapResult, *result);
-- 
2.43.0

